### DISCUSSION TITLE
OAuth2 plugin: Token Introspection support

### DISCUSSION THREAD

**Discussion by vitorsdcs** [Kong/kong#7248] [Ideas and feature requests]
Token Introspection allows APIs to check the state of an OAuth2 token.

More info: https://tools.ietf.org/html/rfc7662

**Comment by subnetmarco** [comment]
@vitorsdcs you can also invoke the `/oauth2_tokens` endpoint on the Admin API to get a list of all the tokens on the system and their properties.

**Comment by p0pr0ck5** [comment]
This was done! :) https://getkong.org/plugins/oauth2-introspection/

**Comment by subnetmarco** [comment]
I assume this issue requested to implement the Introspection endpoint support as a new endpoint in the existing OAuth 2.0 plugin, as opposed to the linked OAuth 2.0 Introspection Endpoint plugin which leverages a third-party Introspection endpoint to authenticate requests.

If that's the case, it should be re-opened.

**Comment by p0pr0ck5** [comment]
Ah yep, I think that's the case. Faulty assumption on my part; thanks.

**Comment by NebojsaKrtolica** [comment]
...so? How's this feature development going? :smiley:
